apps/src/code-studio/pd/application/facilitator1920/Section3ExperienceAndCommitments.jsx

```diff
@@ -0,0 +1,295 @@
+  static labels = PageLabels.section3ExperienceAndCommitments;
+
+  static associatedFields = [
+    ...Object.keys(PageLabels.section3ExperienceAndCommitments),
+    "regionalPartnerId"
+  ];
+
+  state = {
+    loadingFitWorkshops: true,
+    loadingSummerWorkshops: false,
+    partner: null,
+    loadError: false
+  };
+
+  componentDidMount() {
+    this.loadFitWorkshops();
+
+    if ([CSD, CSP].includes(this.props.data.program)) {
+      this.loadSummerWorkshops();
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.loadFitWorkshopsRequest) {
+      this.loadFitWorkshopsRequest.abort();
+    }
+    if (this.loadSummerWorkshopsRequest) {
+      this.loadSummerWorkshopsRequest.abort();
+    }
+  }
+
+  loadFitWorkshops() {
+    const queryParams = {
+      course: this.props.data.program,
+      subject: 'Code.org Facilitator Weekend',
+      zip_code: this.props.data.zipCode,
+      state: this.props.data.state
+    };
+
+    const url = PARTNER_WORKSHOPS_API_ENDPOINT + $.param(queryParams);
+    this.loadFitWorkshopsRequest = $.ajax({
+      method: 'GET',
+      url: url,
+      dataType: 'json'
+    }).done((data) => {
+      this.loadFitWorkshopsRequest = null;
+
+      this.handleChange({
+        regionalPartnerId: data.id,
+        fitWorkshops: data.workshops,
+        regionalPartnerName: data.name
+      });
+
+      this.setState({
+        loadingFitWorkshops: false
+      });
+    }).error(() => {
+      this.setState({
+        loadingFitWorkshops: false,
+        loadError: true
+      });
+    });
+  }
+
+  loadSummerWorkshops() {
+    this.setState({loadingSummerWorkshops: true});
+
+    const queryParams = {
+      course: this.props.data.program,
+      subject: '5-day Summer',
+      zip_code: this.props.data.zipCode,
+      state: this.props.data.state
+    };
+
+    const url = PARTNER_WORKSHOPS_API_ENDPOINT + $.param(queryParams);
+    this.loadSummerWorkshopsRequest = $.ajax({
+      method: 'GET',
+      url: url,
+      dataType: 'json'
+    }).done((data) => {
+      this.loadSummerWorkshopsRequest = null;
+
+      this.handleChange({
+        regionalPartnerId: data.id,
+        summerWorkshops: data.workshops,
+        regionalPartnerName: data.name
+      });
+
+      this.setState({
+        loadingSummerWorkshops: false
+      });
+    }).error(() => {
+      this.setState({
+        loadingSummerWorkshops: false,
+        loadError: true
+      });
+    });
+  }
+
+  renderCsdCspWhichFitWeekend() {
+    if (this.props.data.fitWorkshops && this.props.data.fitWorkshops.length > 0) {
+      const options = this.props.data.fitWorkshops.map(workshop =>
+        `${workshop.dates} in ${workshop.location}`
+      );
+      options.push(TextFields.notSurePleaseExplain, TextFields.unableToAttendPleaseExplain);
+
+      const textFieldMap = {
+        [TextFields.notSurePleaseExplain] : "not_sure",
+        [TextFields.unableToAttendPleaseExplain] : "unable_to_attend"
```

Comment:
"not_sure" and "unable_to_attend" are used in multiple places. Perhaps move to variable